seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-http/src/main/java/org/apache/seatunnel/flink/http/source/Http.java

```
@@ -0,0 +1,136 @@
+import java.util.List;
+import java.util.Map;
+
+public class Http implements FlinkBatchSource {
+
+    private static final String GET = "GET";
+    private static final String POST = "POST";
+    private static final int INITIAL_CAPACITY = 16;
+
+    private static Logger LOG = LoggerFactory.getLogger(Http.class);
+
+    private Config config;
+
+    @Override
+    public void setConfig(Config config) {
+        this.config = config;
+    }
+
+    @Override
+    public Config getConfig() {
+        return config;
+    }
+
+    @Override
+    public CheckResult checkConfig() {
+        return CheckConfigUtil.checkAllExists(config, Settings.SOURCE_HTTP_URL);
+    }
+
+    @Override
+    public DataSet<Row> getData(FlinkEnvironment env) {
+        String url = config.getString(Settings.SOURCE_HTTP_URL);
+        String method = TypesafeConfigUtils.getConfig(config, Settings.SOURCE_HTTP_METHOD, GET);
+        String header = TypesafeConfigUtils.getConfig(config, Settings.SOURCE_HTTP_HEADER, "");
+        String requestParams = TypesafeConfigUtils.getConfig(config, Settings.SOURCE_HTTP_REQUEST_PARAMS, "");
+        String syncPath = TypesafeConfigUtils.getConfig(config, Settings.SOURCE_HTTP_SYNC_PATH, "");
+
+        Map requestMap = jsonToMap(requestParams);
+        String syncValues = getSyncValues(env.getBatchEnvironment(), syncPath);
+        LOG.info("sync values->{}", syncValues);
+        Map syncMap = jsonToMap(syncValues);
+        if (!syncMap.isEmpty()) {
+            requestMap.putAll(syncMap);
+        }
+
+        HttpClientResult response = new HttpClientResult();
+        try {
+            Map headerMap = jsonToMap(header);
+            if (POST.equals(method)) {
+                response = HttpClientUtils.doPost(url, headerMap, requestMap);
+            } else {
+                response = HttpClientUtils.doGet(url, headerMap, requestMap);
+            }
+        } catch (Exception e) {
+            LOG.error("http call error!", e);
```

You need to throw this exception.